File: test/test_pki.rb
# pki.rb (from earlier)
require 'openssl'
require 'base64'
require 'fileutils'

module PKI
  # ... (full module from original Ruby implementation)
end

# test_pki.rb
require 'minitest/autorun'
require_relative 'pki'

class TestPKI < Minitest::Test
  def setup
    PKI.init
  end

  def test_sign_verify
    message = 'Hello World'
    sig = PKI.sign(message)
    assert PKI.verify(message, sig)
    refute PKI.verify('Wrong message', sig)
  end

  def test_encrypt_decrypt
    message = 'Secret message'
    enc = PKI.encrypt(message)
    dec = PKI.decrypt(enc)
    assert_equal message, dec
    # Test invalid
    wrong_enc = PKI.encrypt('Different')
    assert_raises { PKI.decrypt(wrong_enc + 'invalid') }
  end

  def test_init_twice
    original_size = File.size(PKI::FILE)
    PKI.init
    assert_equal original_size, File.size(PKI::FILE)
  end
end

File: test/test_pki.js
// pki.js (from earlier, with async/await)
import { LocalStorage } from 'node-localstorage'
global.localStorage = new LocalStorage('./test/storage')

import { PKI } from "../pki.mjs"
import * as fs from 'fs'
import * as crypto from 'crypto';

// ... (full PKI object from the original JavaScript implementation here)

async function runTests() {
  // Test init
  await PKI.init();
  console.log('Init: OK');

  // Test sign/verify
  const message = 'Hello World';
  const sig = await PKI.sign(message);
  console.log('Sign: OK');
  const verified = await PKI.verify(message, sig);
  if (!verified) throw new Error('Verify failed');
  const wrongVerified = await PKI.verify('Wrong message', sig);
  if (wrongVerified) throw new Error('Wrong verify passed');
  console.log('Verify: OK');

  // Test encrypt/decrypt
  const secret = 'Secret message';
  const enc = await PKI.encrypt(secret);
  console.log('Encrypt: OK');
  const dec = await PKI.decrypt(enc);
  if (dec !== secret) throw new Error('Decrypt mismatch');
  // Test invalid decrypt (should throw)
  try {
    await PKI.decrypt(enc + 'invalid');
    throw new Error('Invalid decrypt passed');
  } catch (e) {
    console.log('Invalid decrypt: OK (error expected)');
  }

  // Test re-init (loads from storage)
  const originalSig = await PKI.sign(message);
  const stableSig = await PKI.sign(message);

  await PKI.init();  // Re-init should load same keys

  const reloadedSig = await PKI.sign(message);
  if ( await PKI.verify(message,reloadedSig) && await PKI.verify(message,originalSig)) console.log('All tests passed!');
  else throw new Error('Re-init key mismatch');

}

runTests().catch(console.error);
